File: XWeather/XWeather/Domain/Wu/Astronomy.cs
using System;
namespace XWeather.Domain
{
	public class Astronomy : WuObject
	{
		public override string WuKey => "astronomy";

		public MoonPhase moon_phase { get; set; }
		public AstronomyPhase sun_phase { get; set; }
	}


	public class MoonPhase : AstronomyPhase
	{
		public string percentIlluminated { get; set; }
		public string ageOfMoon { get; set; }
		public string phaseofMoon { get; set; }
		public string hemisphere { get; set; }
		public AstronomyTime current_time { get; set; }
		public AstronomyTime moonrise { get; set; }
		public AstronomyTime moonset { get; set; }
	}


	public class AstronomyPhase
	{
		public AstronomyTime sunrise { get; set; }
		public AstronomyTime sunset { get; set; }
	}


	public class AstronomyTime
	{
		public double hour { get; set; }
		public double minute { get; set; }

		public DateTime LocalDateTime => DateTime.Today.AddHours (hour).AddMinutes (minute);
	}
}

File: Swift/XWeather/XWeather/Domain/Wu/Astronomy.swift
//
//  Astronomy.swift
//  XWeather
//
//  Created by Colby Williams on 9/13/16.
//  Copyright © 2016 Colby Williams. All rights reserved.
//

import Foundation

public class Astronomy : WuObject {

	public override var WuKey: String { get { return "astronomy" } }

	public var moon_phase: MoonPhase?
	public var sun_phase: AstronomyPhase?
}


public class MoonPhase : AstronomyPhase {

	public var percentIlluminated: String?
	public var ageOfMoon: String?
	public var phaseofMoon: String?
	public var hemisphere: String?
	public var current_time: AstronomyTime?
	public var moonrise: AstronomyTime?
	public var moonset: AstronomyTime?
}


public class AstronomyPhase {

	public var sunrise: AstronomyTime?
	public var sunset: AstronomyTime?
}


public class AstronomyTime {

	public var hour: Double?
	public var minute: Double?

	//public DateTime LocalDateTime => DateTime.Today.AddHours (hour).AddMinutes (minute);
}
